### PR TITLE
Feature/devise auth

### PR DESCRIPTION
**Pull request by ulisesedu7**
## Description
In this pull request, I've installed the devise gem and set up user log-in authentication to my blog app.

**Task:** https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md

This PR makes the following changes:
- Installed devise gem
- user rails generate to install the views and controllers updates to my user's authentication setup.

**Note: David and I joined a zoom call to work on both of our repositories at the same time, we started working side by side so we didn't commit on each's repositories, that's why we don't have commits, but we did work together and shared code through zoom.**

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

## Checklist

- [X] My code follows the style guidelines of this project (no linters errors)
- [X] I have checked my code and corrected any misspellings
- [X] I have commented on my code
- [X] I have deleted any unused code as comments in my codebase

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @Team,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have commits from both partners, based on the [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_devise.md#work-and-submission-mode) you should both  contribute to this project 👍🏿 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) do |u|
+      u.permit(:name, :email, :password, :password_confirmation)
+    end
+    devise_parameter_sanitizer.permit(:account_update) do |u|
+      u.permit(:name, :email, :password, :password_confirmation, :current_password)
+    end
```
Comment (TedLivist) [inline app/controllers/application_controller.rb]: - [x] When I visit your root path, your app displays the image below. But that should not be so. Devise provides a method for ensuring that only logged in users can navigate the app. Kindly add this helper method to this `application_controller`:

  ```
  before_action :authenticate_user!
  ```
  This way when a user is not logged they are redirected to the sign in page

  fwel

File: app/views/devise/passwords/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :post }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Send me reset password instructions" %>
```
Comment (TedLivist) [inline app/views/devise/passwords/new.html.erb]: - [x] When I try to request a new password I get this error. You can fix this by disabling `turbo` like so:

  ```
  <%= f.submit "Send me reset password instructions", data: { turbo: false } %>
  ```

  djdk

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (TedLivist) [inline app/views/devise/registrations/new.html.erb:34]: - [x] Kindly style your app to match the wireframe. From the sneak peek the form should be somewhat centered. Kindly do the same

  kilf

File: app/views/devise/sessions/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (TedLivist) [inline app/views/devise/sessions/new.html.erb:26]: - [x] Kindly style your app to match the wireframe. From the sneak peek the form should be somewhat centered. Kindly do the same

  logp

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @ulisesedu7 and @David-Verde,

Good job so far!
✔️ Good documentation
✔️ Implementation of sign up and sign in with Devise
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ulisesedu7** [comment]
@TedLivist Hi Ted, thank you very much for all the feedback, I'll start working on it right now.

File: app/views/devise/passwords/new.html.erb (hunk 2 above)
Comment (ulisesedu7) [reply app/views/devise/passwords/new.html.erb]: @TedLivist Awesome, good catch there, thank you very much for it.

**Review by Munsa1 — APPROVED** [review]
Hi @ulisesedu7, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**